extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java

```diff
@@ -432,15 +434,15 @@ private static void contributeMethodsWithInheritedValidation(
     }
 
     private static Map<DotName, Set<SimpleMethodSignatureKey>> gatherJaxRsMethods(
-            List<AdditionalJaxRsResourceMethodAnnotationsBuildItem> additionalJaxRsResourceMethodAnnotations,
+            Set<DotName> additionalJaxRsResourceMethodAnnotations,
             IndexView indexView) {
         Map<DotName, Set<SimpleMethodSignatureKey>> jaxRsMethods = new HashMap<>();
 
         Collection<DotName> jaxRsMethodDefiningAnnotations = new ArrayList<>(
                 ResteasyDotNames.JAXRS_METHOD_ANNOTATIONS.size() + additionalJaxRsResourceMethodAnnotations.size());
         jaxRsMethodDefiningAnnotations.addAll(ResteasyDotNames.JAXRS_METHOD_ANNOTATIONS);
-        for (AdditionalJaxRsResourceMethodAnnotationsBuildItem additionalJaxRsResourceMethodAnnotation : additionalJaxRsResourceMethodAnnotations) {
-            jaxRsMethodDefiningAnnotations.addAll(additionalJaxRsResourceMethodAnnotation.getAnnotationClasses());
+        for (DotName additionalJaxRsResourceMethodAnnotation : additionalJaxRsResourceMethodAnnotations) {
+            jaxRsMethodDefiningAnnotations.add(additionalJaxRsResourceMethodAnnotation);
         }
```

Comment:
I know it's a detail, but if you're going to amend the commit anyway...

```suggestion
        jaxRsMethodDefiningAnnotations.addAll(additionalJaxRsResourceMethodAnnotations);
```